### ISSUE TITLE
[All] product: standard_price should be 'Actual' Cost' not 'COGS'

### ISSUE THREAD

**Issue by gustavovalverde** [odoo/odoo#26321]
#### Impacted versions:
All

#### Current behavior:
The cost shown on the product is the financial Cost registered **after** the product was sold (**COGS**). Which is an information needed on a P&L, but it's confusing on the day-to-day use and it also causes computation issues on a Product Life-cycle inside a company before the product is sold, mainly on **FIFO** but it also has impact on **AVCO** on specific scenarios.

**For example:**
- Pricelist Computation **based on Cost** won't work until the first product is sold: https://github.com/odoo/odoo/issues/25977
- A **just created** product can't be Manufactured if a standard cost is not specified in the product template (but there's no cost, not until its first manufacturing): https://github.com/odoo/odoo/issues/25290
- **Raw materials** with FIFO that are not sold (which won't be) are not considered in the BOM computation: https://github.com/odoo/odoo/issues/26169
- **Margins** can't be calculated if there's not a first sale (so the computation won't work until the first product is sold): https://github.com/odoo/odoo/issues/25604

#### Expected behavior:
Basically, as a bunch of computations in Odoo are based on the `standard_price`, this cost should be the **actual cost,** and not the **cost of goods sold (COGS)** this way computations that depend on the cost are correctly done, before the product is sold, when using FIFO as a costing method.

This will also reduce all the confusion that it's generated to final users when they buy something and can't see the cost of the product.

**Note:** This is all **mainly** happening on FIFO, most of this computations are OK on AVCO. And it's mainly because most of them are done using the `standard_price` and not the `stock_value`.

**Comment by nhomar** [comment]
Did you try setting the first cost when you just create the product itself? with that functional approach this is solved immediately in all your three cases.

But IMHO this is a real issue for now and is a little tricky and not obvious, the basic and most obvious behavior should be set the very first cost at the moment of a purchase and/or billing and/or reception (you should decide which document rules your use case).

BTW this second paragraph can be solved with a very simple automated action (but yes it is not ideal) that check if it is the very first document set the cost itself.

LEt's see what Odoo say.

@lap-odoo this case was discussed by us few time ago when the new design arrive, can you help with your PoV please?

**Comment by nhomar** [comment]
I forget mention something:

BTW: **'Actual' Cost'** is useless, because such value is not the same per every single move, that's why all the FIFO logic exists by itself.

Regards.

**Comment by gustavovalverde** [comment]
> Did you try setting the first cost when you just create the product itself? with that functional approach this is solved immediately in all your three cases.

Let's call this approach: **Manual Cost**

That's a workaround, but not a solution. In most of the examples above this would throw **far from real information:**
- Pricelist Computation **based on Cost** and Margins won't consider additional costs like landed costs
- In MRP a **just created** product will use _Manual Cost_ instead of the sum of the raw materials used in the production.
- **Raw materials** will keep the _Manual Cost_ not matter the cost fluctuation until an out_move is created.

**Comment by gustavovalverde** [comment]
IMHO, this cost shouldn't be set manually on a day to day operation. There's already a bunch of processes going on: Purchase, Reception, Landed Cost, Billing. Each one of then needing a product cost to work correctly.

So why is all of this happening and yet the `standard_cost` stays empty? _Flawing_ all computations that depends product.template standard_cost.

**Comment by nhomar** [comment]
@gustavovalverde 

You need seriously my friend to document yourself on this point a little bit.

What you called **Manual Cost** is possible and it is called [**Standard Costing**](https://www.accountingtools.com/articles/2017/5/14/standard-costing).

**From the link:**

>Standard costing is the practice of substituting an expected cost for an actual cost in the accounting records, and then periodically recording variances showing the difference between the expected and actual costs. This approach represents a simplified alternative to cost layering systems, such as the FIFO and LIFO methods, where large amounts of historical cost information must be maintained for items held in stock.

I will point you out to the phrase that conceptually will help you:

**and then periodically recording variances showing the difference between the expected and actual costs**

Then what you need to set you products as **Standard** no FIFO, no Average, **Standard**.

And then set a proper logic throught either a cronjob or a server action that allows to you to set the proper cost writting the product.

Let's say an automated action that write the number from the origin you want the cost will com from every time you set or validate the document you decide in your business model.

Odoo itself uses **Financial FIFO** and that's not what you need.

You can set also the average cost but that's not what you can use to set the price based on cost.

- IF you want to set your pricelist based on costs you **must** use Standard cost and automate your criteria through a customization.
- If you want to base your pricelists in average cost, then you live in an stable economy (which is not common on latam).
- If you decide use FIFO and use pricelist based on costs, then you must live with the fact that you need a little automation to set the first cost and/or set the the very first cost in your PLM. 

I hope it helps to your use case, and help you to solve the problem, but by far what odoo have at the end now is really good and need improvements, (may be on master set the very first cost can help) but I think that most of companies set this manually.

**Foot note:** Let me tell you that I walk that road also fighting against the right design, and this is not a bug (and I had this fights internally also), please give the benefit of the doubt check the information I am sharing and come back with more precise issues if you find some. [Recommended lecture.](https://www.principlesofaccounting.com/) This is not a technical issue but a conceptual one.
